mods/ra/weapons/other.yaml

```diff
@@ -177,6 +208,22 @@ TTankZap:
 			None: 1000
 		DamageTypes: Prone50Percent, TriggerProne, ElectricityDeath
 
+TTankCharge:
+	ReloadDelay: 120
+	Range: 2c0
+	Report: TESLACHARGE.AUD
+	Charges: yes
+	ValidStances: Ally
+	ValidTargets: TeslaBoost
+	Projectile: TeslaZap
+		Image: litningupg
+	Warhead@charge: GrantUpgrade
+		Range: 42
+		Duration: 125
+		Upgrades: tesla, tesla1, tesla2
```

Comment:
I've made the `UpgradeManager` support requests to grant a timed upgrade multiple times from a single source. The maximum number of times it will grant needs to be supplied by the caller (in this instance `GrantUpgradeWarhead`), otherwise it is assumed to be 1.

What `GrantUpgradeWarhead` then does when it gets `Upgrades == { "tesla", "tesla", "tesla" }`, it requests granting of each of these tesla ups, but notifying that it should be granted up to three times from this particular source.

How does this sound?